tests/drivers/memc/smc_sram/testcase.yaml

```diff
@@ -0,0 +1,5 @@
+tests:
+  drivers.memc.smc_sram:
+    tags: drivers memc
+    depends_on: memc
+    filter: dt_compat_enabled("atmel,sam-smc")
```

Comment:
Another option would be to put this under `tests/boards/atmel_sam` or similar - of make it generic for all memc drivers.

Reply:
This suggestion was already done by @nandojve , but I wanted to this in a separate PR.

Reply:
Why not fix it before introducing it?

Reply:
note: `memc` are a set of drivers without an API, so a generic test isn't possible afaik.

Reply:
The current test simply tests RW access to a memory region from a device on the memory controller - that sounds like something that could work for at least more than one memc device type.

Reply:
@henrikbrixandersen ok, to update the test in a follow-up PR?

Reply:
I don't see why we should introduce the test with identified issues only to do a follow-up PR. Please address it in this PR.

Reply:
Done